syft/cataloger/catalog.go

```diff
@@ -39,6 +39,8 @@ func newMonitor() (*progress.Manual, *progress.Manual) {
 // request.
 func Catalog(resolver source.Resolver, theDistro *distro.Distro, catalogers ...Cataloger) (*pkg.Catalog, error) {
 	catalog := pkg.NewCatalog()
+	defer catalog.Finalize()
```

Comment:
I'd suggest changing this from using `defer` to just calling the method explicitly at the right time later on in the implementation. If I'm understanding correctly, it's not universally true that we want this to run before the function returns — specifically if we're returning an error instead of the catalog